Add value `ingress.hosturl` for externally provided ingress

rconway opened the pull request:
This solves the problem of how to provide the URL of the ZOO service in the case that the ingress is provided externally - i.e. `ingress.enabled: false` with the ingress not being provided by the helm chart, but instead by other means within the deployment. For example, in the case that a custom IAM approach is externally configured on the request path to the ZOO service.

Before this fix, the current behaviour is that the service URL is deduced from `ingress.hosts.host[0]` if ingress is configured in the helm chart - otherwise it is set as `http://localhost:8080`.

This fix introduces helm tpl helper functions `zoo-project-dru.hosturl` (HTTP) and `zoo-project-dru.hosturlws` (Web Sockets) to deduce the service URL in the following order of priority...
* `ingress.hosturl`
* `ingress.hosts.host[0]` if `ingress.enabled: true`
* `http://localhost:8080` if all else fails

gfenoy commented:
Great fix, thank you.

Could you update the chart version as well, please?

I was wondering if we should add a dedicated ingress section in the [README.md](https://github.com/ZOO-Project/charts/blob/main/zoo-project-dru/README.md) file too.

rconway commented:
Will update the chart version - didn't want to presume to do this.
Also, I'll have a look at the README for the ingress.

rconway commented:
@gfenoy 
See additional commits for...
* Updated chart version and added `Chart.lock` file. It is understood to be best practice to include the Chart.lock file in the chart in order to establish version consistency
* Updated README for ingress

gfenoy commented:
Thanks for the clarification about the `Chart.lock` file and for including the dedicated ingress section in the `README.md`.
